exercise/exercise_models.py

```diff
@@ -694,7 +695,16 @@ def grade(self, request, submission, no_penalties=False, url_name="exercise"):
         """
         Loads the exercise feedback page.
         """
-        language = get_language()
+        # Get the language from the submission
+        try:
+            aplus_data_index = [i[0] for i in submission.submission_data].index("__aplus__")
```

Comment:
`_aplus_group` should then also be moved to the submission meta data from the normal submission data. At least the new submit API should support it before the new API is widely used. The old ExerciseView could be changed later.

The old hack `__grader_lang` possibly can't be changed now, or at least it would also require changes in the MOOC-Grader too.

Reply:
Once A+ handles the language correctly, then MOOC-Grader can possibly drop that field. In other words, leave the `__grader_lang` to be MOOC-Grader's problem.

Reply:
> What kind of data would be included in the meta? Only the language?

For now, nothing else than lang and group id. It makes it easy place to start adding values.

Reply:
Good idea might be a bit less hacky than this:), should the submission_data stay as it is even it means there would be some duplicate data

Reply:
No, we don't want to have the new `__aplus__` field in the submission data. Its contents shall be stored in the new submission meta data field in the Submission model. The database field does not need to contain the word `__aplus__` as it is enough to store the actual JSON object. This object is stored as an object so no need to convert it into a list of 2-tuples like has previously been done with the submission data.

Reply:
About moving `_aplus_group` into the new submission meta data: this line in `check_submission_allowed` uses the old way without the new `__aplus__` object: https://github.com/apluslms/a-plus/blob/2195e36b86db9eb77c6695fdb37c5af7bba4dad9/exercise/exercise_models.py#L578
The function is called by the normal web submission (`class ExerciseView`) and the new submit API endpoint. If we don't modify the web submissions now, then we need to support both ways. The API should prefer the new method with `__aplus__` since the API itself is new. Note that `check_submission_allowed` does not have any `submission` object since it is called before creating a new submission, but then we can assume that the `request.POST` contains the values.